### PR TITLE
CLI | Change Vorpal vorpal-latest-version flag flow (AST-48376)

### PR DESCRIPTION
**Pull request by AlvoBen**
By submitting a PR to this repository, you agree to the terms within the [Checkmarx Code of Conduct](../docs/code_of_conduct.md). Please see the [contributing guidelines](../docs/contributing.md) for how to create and submit a high-quality PR for this repo.

### Description

> change vorpal-latest-version flag flow

### References

> https://checkmarx.atlassian.net/browse/AST-48376

### Testing

> Added unit tests

### Checklist

- [x] I have added documentation for new/changed functionality in this PR (if applicable).
- [x] I have updated the CLI help for new/changed functionality in this PR (if applicable).
- [x] All active GitHub checks for tests, formatting, and security are passing
- [x] The correct base branch is being used

### CODE REVIEW

File: internal/services/vorpal.go [hunk 1]
```diff
@@ -93,16 +93,22 @@ func executeScan(vorpalWrapper grpcs.VorpalWrapper, filePath string) (*grpcs.Sca
...
 
-func manageVorpalInstallation(vorpalParams VorpalScanParams, vorpalWrapper grpcs.VorpalWrapper) error {
+func manageVorpalInstallation(vorpalParams VorpalScanParams, vorpalWrapper grpcs.VorpalWrapper, vorpalWrappers VorpalWrappersParam) error {
 	vorpalInstalled, _ := osinstaller.FileExists(vorpalconfig.Params.ExecutableFilePath())
 
-	if vorpalParams.VorpalUpdateVersion || !vorpalInstalled {
-		if err := vorpalWrapper.HealthCheck(); err == nil {
-			_ = vorpalWrapper.ShutDown()
+	if !vorpalInstalled || vorpalParams.VorpalUpdateVersion {
+		if err := checkLicense(vorpalParams.IsDefaultAgent, vorpalWrappers); err != nil {
```
Comment (tamarleviCm) [inline internal/services/vorpal.go:100]: I think Or told us to shut down the service here

File: internal/services/vorpal.go [hunk 2]
```diff
@@ -93,16 +93,22 @@ func executeScan(vorpalWrapper grpcs.VorpalWrapper, filePath string) (*grpcs.Sca
...
+			return err
 		}
-		if err := osinstaller.InstallOrUpgrade(&vorpalconfig.Params); err != nil {
+		newInstallation, err := osinstaller.InstallOrUpgrade(&vorpalconfig.Params)
+		if err != nil {
 			return err
 		}
+		if newInstallation && vorpalParams.VorpalUpdateVersion {
```
Comment (tamarleviCm) [inline internal/services/vorpal.go:107]:
```suggestion
		if newInstallation {
```
in a situation we check together that I removed the executable file- we also want to shut down the service

**Review by tamarleviCm — APPROVED** [review]

File: internal/services/osinstaller/os-installer.go [hunk 3]
```diff
@@ -50,38 +50,38 @@ func downloadFile(downloadURLPath, filePath string) error {
 // InstallOrUpgrade Checks the version according to the hash file,
 // downloads the RealTime installation if the version is not up-to-date,
 // Extracts the RealTime installation according to the operating system type
-func InstallOrUpgrade(installationConfiguration *InstallationConfiguration) error {
+func InstallOrUpgrade(installationConfiguration *InstallationConfiguration) (bool, error) {
```
Comment (OrShamirCM) [inline internal/services/osinstaller/os-installer.go:53]: is it possible to add an alias to the bool? like - bool doesInstalltion

File: internal/services/vorpal.go [hunk 4]
```diff
@@ -93,16 +93,21 @@ func executeScan(vorpalWrapper grpcs.VorpalWrapper, filePath string) (*grpcs.Sca
...
 			_ = vorpalWrapper.ShutDown()
+			return err
 		}
-		if err := osinstaller.InstallOrUpgrade(&vorpalconfig.Params); err != nil {
+		newInstallation, err := osinstaller.InstallOrUpgrade(&vorpalconfig.Params)
+		if err != nil {
 			return err
 		}
+		if newInstallation {
+			_ = vorpalWrapper.ShutDown()
```
Comment (OrShamirCM) [inline internal/services/vorpal.go:109]: we are doing a shutdown in other flow before we are running a new one no? just need to make sure there are no bugs in this flow.
let's have tests please..we have many scenarios

Comment (AlvoBen) [reply internal/services/vorpal.go:109]: This function is the only place we are shutting down. This pr did not change this logic

Comment (OrShamirCM) [reply internal/services/vorpal.go:109]: we didn't have shutdown aywehre else?

File: internal/services/vorpal.go (hunk 1 above)
Comment (OrShamirCM) [reply internal/services/vorpal.go:100]: we need to check here for helth check before shutting down. or the shut down check that there sis engine alive bofre>?

File: internal/services/vorpal.go (hunk 4 above)
Comment (OrShamirCM) [reply internal/services/vorpal.go:109]: same here for the shutdown and healthcheck

Comment (AlvoBen) [reply internal/services/vorpal.go:109]: No, this is the only function that we use shutdowns. 
For healthcheck there are more places

**Review by OrShamirCM — APPROVED** [review]

**Pull request merged**